doc/NPCs.md

```diff
@@ -1,4 +1,54 @@
 TODO: document the "npc" structure, used to load NPC template
+# Creating new NPCs
+Often times you will want to create new NPCs to either add quests, flavor or to introduce little known activities to the larger player base. New NPCs are written entirely in JSON so they are one of the easier places to begin your contributions.
+
+There are two parts to creating a new NPC separate of any dialogue you may choose to add.  
+First there is the `npc_class` which follows the following template.
+Format:
+```json
+{
+  "type": "npc_class",
+  "id": "NC_EXAMPLE",
+  "name": { "str": "Example NPC" },
+  "job_description": "I'm helping you learn the game.",
+  "common": false,
+  "bonus_str": { "rng": [ -4, 0 ] },
+  "bonus_dex": { "rng": [ -2, 0 ] },
+  "bonus_int": { "rng": [ 1, 5 ] },
+  "skills": [
+    {
+      "skill": "ALL",
+      "level": { "mul": [ { "one_in": 3 }, { "sum": [ { "dice": [ 2, 2 ] }, { "constant": -2 }, { "one_in": 4 } ] } ] }
+    }
+  ],
+  "worn_override": "NC_EXAMPLE_worn",
+  "carry_override": "NC_EXAMPLE_carried",
+  "weapon_override": "NC_EXAMPLE_weapon",
+  "shopkeeper_item_group": "example_shopkeeper_itemgroup",
+  "traits": [ { "group": "BG_survival_story_EVACUEE" }, { "group": "NPC_starting_traits" }, { "group": "Appearance_demographics" } ]
+}
+```
+There are a couple of items in the above template that may not be self evident.  `"common": false` means that this npc class will not spawn randomly, it defaults to true if unlisted. `"shopkeeper_item_group"` is only needed if the planned NPC will be a shopkeeper with a revolving stock of items that change every three in game days. All of the item overrides will ensure that any NPC of this class spawns with specific items.
+There is a second template required for a new NPC. It looks as follows.
+Format:
+```json
+{
+  "type": "npc",
+  "id": "examplicious",
+  "//": "The luckiest NPC to never experience the Cataclysm.",
+  "name_suffix": "examplar",
+  "class": "NC_EXAMPLE",
+  "attitude": 0,
+  "mission": 7,
+  "chat": "TALK_EXAMPLE",
+  "faction": "no_faction"
+}
+```
+This is the json that creates the NPC ID that is used to spawn an NPC in mapgen.
+Attitude is based on the enum in npc.h. The important ones are 0=NPCATT_NULL, 1=NPCATT_TALK", 3=NPCATT_FOLLOW, 7=NPCATT_DEFEND, 10=NPCATT_KILL, and 11=NPCATT_FLEE.
```

Comment:
```suggestion
Attitude is based on the enum in [src/npc.h]. The important ones are:
* [NPCATT_NULL](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L87) (0)
* [NPCATT_TALK](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L88) (1)
* [NPCATT_FOLLOW](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L90) (3)
* NPCATT_DEFEND (7)
* [NPCATT_KILL](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L97) (10)
* [NPCATT_FLEE](https://github.com/CleverRaven/Cataclysm-DDA/blob/931c0779f36762e73374f6ae053ec2617cf1da9b/src/npc.h#L98) (11)

Reply:
The original text is from the notes for an NPC in the refugee center.